File: backend/app/schemas/timesheet.py
from datetime import datetime
from typing import Optional, List

from pydantic import BaseModel


class TimesheetUserInfo(BaseModel):
    name: str
    email: str


class TimesheetOut(BaseModel):
    timesheet_id: int
    user_id: str
    client_name: Optional[str] = None
    project_name: Optional[str] = None
    work_date: datetime
    type_of_day: str
    hours_worked: Optional[float] = None
    comments: Optional[str] = None
    status: str
    manager_reason: Optional[str] = None
    updated_at: datetime
    user: Optional[TimesheetUserInfo] = None

    class Config:
        from_attributes = True


class TimesheetCreate(BaseModel):
    client_name: Optional[str] = None
    project_name: Optional[str] = None
    work_date: str
    type_of_day: str = "Working"
    hours_worked: Optional[float] = None
    comments: Optional[str] = None


class ManagerGroupedResponse(BaseModel):
    user_id: str
    full_name: str
    email: str
    entries: List[TimesheetOut]


class ClientSubmissionOut(BaseModel):
    id: int
    user_id: str
    client_name: str
    client_manager_name: str
    client_manager_email: str
    from_date: datetime
    to_date: datetime
    submitted_at: datetime
    approval_token: str
    cs_status: str
    responded_at: Optional[datetime] = None
    rejection_note: Optional[str] = None
    user: Optional[TimesheetUserInfo] = None

    class Config:
        from_attributes = True


class ClientSubmissionPayload(BaseModel):
    client_name: str
    client_manager_name: str
    client_manager_email: str
    from_date: str
    to_date: str
    csv_content: str

File: frontend/src/types/timesheet.ts
export type DayType = 'Working' | 'Leave' | 'Holiday' | 'HalfDay' | 'CompOff'
export type UserRole = 'EMPLOYEE' | 'ADMIN'

export interface TimesheetEntry {
  timesheet_id: number
  user_id: string
  client_name: string | null
  project_name: string | null
  work_date: string
  type_of_day: DayType
  hours_worked: number | null
  comments: string | null
  created_at: string | null
  updated_at: string
  user?: {
    name: string
    email: string
  }
}

export interface CreateTimesheetPayload {
  client_name?: string
  project_name?: string
  work_date: string
  type_of_day: DayType
  hours_worked?: number
  comments?: string
}

export interface ManagerGroupedResponse {
  user_id: string
  full_name: string
  email: string
  entries: TimesheetEntry[]
}

export interface ClientSubmissionPayload {
  client_name: string
  client_manager_name: string
  client_manager_email: string
  from_date: string
  to_date: string
  csv_content: string
}

export interface MockUser {
  id: string
  name: string
  email: string
  employeeId: string
  managerId: string
  managerName: string
  designation: string
  leave_balance: number
}
